shoop/core/models/addresses.py

```diff
@@ -173,6 +140,50 @@ def as_string_list(self, locale=None):
     def __iter__(self):
         return iter(self.as_string_list())
 
+    def immutable(self, ignores=()):
```

Comment:
All in all, the semantics of the `immutable()` and `new_mutable()` (which have confusing names if you ask me) should be better documented. For instance, as I see it, this method may sometimes save an all new address? Maybe?

Also, I see no test cases focusing particularly on these new APIs; even if they might be covered by other tests, I'd still like to see them being explicitly tested.